scripts/announcements.R

```diff
@@ -0,0 +1,192 @@
+            "
+        )
+
+    .data %>%
+        mutate(content = content, title = str_c(title, " - ", day_month(date, add_name = FALSE))) %>%
+        select(title, content, skill_level, gh_labels)
+}
+
+create_gh_issues <- function(.data) {
+    .data %>%
+        gh_issue_info() %>%
+        pmap( ~ post_gh_issue(..1, ..2, c(..3, ..4)))
+}
+
+create_gh_issues(new_sessions)
+
+# Create files in _posts/ -------------------------------------------------
+# Adds the new sessions/events to the _posts folder.
+
+create_new_posts_with_content <- function(.data) {
+    new_post_filenames <-
+        glue_data(new_sessions, "{here::here('_posts')}/{date}-{key}.md")
+
+    # Get the GitHub Issue URL for the event.
+    gh_issue_number <- gh::gh("GET /repos/:owner/:repo/issues",
+                              owner = "uoftcoders",
+                              repo = "Events") %>%
+        map_dfr(~ data_frame(title = .$title, url = .$html_url)) %>%
+        mutate(title = str_remove(title, " - .* [1-9]?$"))
+
+    new_post_content <- .data %>%
+        left_join(gh_issue_number, by = "title") %>%
+        glue_data(
+            '
+            ---
+            title: "{title}"
+            text: "{description}"
+            location: "MADLab at Gerstein"
+            link: "{url}"
+            date: "{as.Date(date)}"
+            startTime: "{start_time}"
+            endTime: "{end_time}"
+            ---
+            '
+        )
+
+    # Save post content to file
+    fs::dir_create(here::here("_posts"))
+    map2(new_post_content, new_post_filenames, ~ write_lines(x = .x, path = .y))
+    usethis:::done("Markdown posts created in _posts/ folder.")
+    return(invisible())
+}
+
+create_new_posts_with_content(new_sessions)
+
+# Create emails for sessions ----------------------------------------------
+
+create_new_emails_for_session <- function(.data) {
+    email_content <- new_sessions %>%
+        mutate(needs_packages = ifelse(
+            !is.na(packages),
+            str_c(
+                "Please also install these packages: ",
+                str_replace_all(packages, " ", ", "), "."
+            ),
+            ""
+        )) %>%
+        mutate_at(vars(start_time, end_time), funs(strftime(., format = "%H:%M"))) %>%
+        glue_data(
+            "
+            <add this to the email subject> {title} - {day_month(date)}
+
+            {description}
+
+            **When**: {day_month(date)}, from {start_time}-{end_time}
+            **Where**: MADLab at [Gerstein Science Information Centre](https://goo.gl/maps/2916Y54jQkx)
+            **Skill level**: {skill_level}
+            **What to bring:**
+
+            *Installation instructions*: You will need to install the appropriate programs. See the {program_language} section of the [installation instructions page](https://au-oc.github.io/content/installation). {needs_packages}
+            "
```

Comment:
I know there is an end note to the bottom of the email that is sent out... is that automatically added on the Gmail account? I can add it here if need be :)

Reply:
Also, need the link to installation instructions (if there are any).